src/main/java/com/google/sps/ChatWebSocket.java

```diff
@@ -59,9 +64,46 @@ public void onOpen(Session session) throws IOException {
     }
 
     @OnMessage
-    public void textMessage(Session session, String msg) {
+    public void textMessage(Session session, String msg) throws IOException {
+        JSONObject messageJson = new JSONObject(msg);
+
+        switch (messageJson.getString("type")) {
+            case "MSG_SEND":
+                chatMessage(session, messageJson);
+                break;
+            case "MAP_SEND":
+
+                break;
+            default:
+                System.out.println("Invalid message");
+        }
+    }
+
+    private void chatMessage(Session session, JSONObject messageData)
```

Comment:
It's not clear from the method name what this does. As a rule of thumb, method name should start with a verb to show what they do. For example, this could be `sendChatMessage`